metadata-ingestion/src/datahub/ingestion/source/state_provider/datahub_ingestion_checkpointing_provider.py

```diff
@@ -42,7 +42,9 @@ def create(
         elif config_dict is None:
             raise ConfigurationError("Missing provider configuration.")
         else:
-            provider_config = DatahubIngestionStateProviderConfig.parse_obj(config_dict)
+            provider_config = (
+                DatahubIngestionStateProviderConfig.parse_obj_allow_extras(config_dict)
```

Comment:
@hsheth2, Thank you for the fix! I'm wondering since `max_thread` is also used in `DatahubRestSinkConfig`, which I think the one that handles the rest sink, should `parse_obj_allow_extras` also be called when creating `DatahubRestSinkConfig`?

Reply:
That one actually should be fine as-is, since that class has a max threads option

The main fix here is the removal of that pydantic field removed validator, which was deleting the user provided max threads config before it got parsed

Reply:
Thank you!